File: pending/121.swift
// 121. Best Time to Buy and Sell Stock
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/description/

class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        let n: Int = prices.count
        var buy: Int = prices[0]
        var answer: Int = 0

        for i in 1..<n {
            let p = prices[i]
            if buy > p {
                buy = p
            }
            answer = max(answer, p - buy)
        }
        return answer
    }
}

File: pending/121.py
# 121. Best Time to Buy and Sell Stock
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock/description/

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        answer = 0
        min_price = prices[0]

        for price in prices[1:]:
            answer = max(answer, price - min_price)
            min_price = min(min_price, price)
        return answer
